src/api.rs

```diff
@@ -146,10 +146,10 @@ impl<L: Language> fmt::Display for SyntaxElement<L> {
 }
 
 impl<L: Language> SyntaxNode<L> {
-    pub fn new_root(green: GreenNode) -> SyntaxNode<L> {
+    pub fn new_root(green: Arc<GreenNode>) -> SyntaxNode<L> {
```

Comment:
API wise, it might be a good idea to hide the fact that we use `Arc` internally, and just document that `GreenNode` is cheap to clone.